OpenRA.Mods.Common/Traits/Buildings/RepairableBuilding.cs

```diff
@@ -150,11 +153,50 @@ void ITick.Tick(Actor self)
 				// The cost is the same regardless of the amount of people repairing
 				var hpToRepair = Math.Min(Info.RepairStep, health.MaxHP - health.HP);
 
-				// Cast to long to avoid overflow when multiplying by the health
-				var cost = Math.Max(1, (int)(((long)hpToRepair * Info.RepairPercent * buildingValue) / (health.MaxHP * 100L)));
+				// Have all cost values multiplied by 100, cost tends to be miniscule. 100x comes from Info.RepairPercent
+				// Cast to long to avoid overflow
+				var cost = (int)((long)hpToRepair * Info.RepairPercent * buildingValue / health.MaxHP);
 
-				// TakeCash will return false if the player can't pay, and will stop him from contributing this Tick
-				var activePlayers = Repairers.Count(player => player.PlayerActor.Trait<PlayerResources>().TakeCash(cost, true));
+				// We are using the highest possible repair cost to find the players that
+				// can pay it as we cannot know what the actual cost will be
+				var maxMultiplier = 0;
+				for (var cnt = 0; cnt < Repairers.Count; cnt++)
+				{
+					var multiplier = Info.RepairBonuses[cnt] / (cnt + 1);
```

Comment:
This also seems to be the only place we divide the repair bonus by the amount of repairers?

Reply:
See lines 170 and 180